src/plugins/dashboard/public/dashboard_app/top_nav/add_panel_action_menu_items.ts

```diff
@@ -5,13 +5,45 @@
  * in compliance with, at your election, the Elastic License 2.0 or the Server
  * Side Public License, v 1.
  */
-import type { ActionExecutionContext, Action } from '@kbn/ui-actions-plugin/public';
+
+import {
+  type ActionExecutionContext,
+  type Action,
+  addPanelMenuTrigger,
+} from '@kbn/ui-actions-plugin/public';
 import { PresentationContainer } from '@kbn/presentation-containers';
-import type {
-  EuiContextMenuPanelDescriptor,
-  EuiContextMenuPanelItemDescriptor,
-} from '@elastic/eui';
-import { addPanelMenuTrigger } from '../../triggers';
+import { COMMON_EMBEDDABLE_GROUPING } from '@kbn/embeddable-plugin/public';
+import type { IconType, CommonProps } from '@elastic/eui';
+import React, { type MouseEventHandler } from 'react';
+
+export interface PanelSelectionMenuItem extends Pick<CommonProps, 'data-test-subj'> {
+  id: string;
+  name: string;
+  icon: IconType;
+  onClick: MouseEventHandler;
+  description?: string;
+  isDisabled?: boolean;
+  isDeprecated?: boolean;
+}
+
+export type GroupedAddPanelActions = Pick<
+  PanelSelectionMenuItem,
+  'id' | 'isDisabled' | 'data-test-subj'
+> & {
+  title: string;
+  items: PanelSelectionMenuItem[];
+};
+
+export interface PlacementPriority {
```

Comment:
I thought we removed this `placementPriority` concept in favour of the `order` key? If so, can we clean these up? If they are still used, I'd like to move away from them because this info should be captured in the normal action types.

Reply:
I removed all references to `placementPriority` in [910065f](https://github.com/elastic/kibana/pull/183764/commits/910065f1055b218450ebd03b92acf0fc2a845ffc).